python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1862,6 +1863,9 @@ def should_start_background_run_thread(self) -> bool:
         """
         return False
 
+    def check_can_launch_step(self, _step: "ExecutionStep") -> bool:
```

Comment:
maybe we should make this a batch request from the get go

Reply:
ah yep, see above. i think its likely we will.